tests/python/pants_test/engine/test_rules.py

```diff
@@ -677,41 +674,53 @@ def b_from_d(d):
                                     subgraph)
 
   def test_validate_yield_statements(self):
-    expected_rx_str = re.escape("""\
-Yield(value=Call(func=Name(id='A', ctx=Load()), args=[], keywords=[]{}))
-""".format('' if PY3 else ', starargs=None, kwargs=None'))
-    with self.assertRaisesRegexp(_RuleVisitor.YieldVisitError, expected_rx_str):
+    with self.assertRaisesRegexp(_RuleVisitor.YieldVisitError, re.escape('yield A()')):
       @rule(A, [])
       def f():
         yield A()
-        # The yield statement isn't at the end.
+        # The yield statement isn't at the end of this series of statements.
+        return
+
+    with self.assertRaises(_RuleVisitor.YieldVisitError) as cm:
+      @rule(A, [])
+      def h():
+        yield A(
+          1 + 2
+        )
         return
+    # Test that the full indentation of multiple-line yields are represented in the output.
+    self.assertIn("""\
+        yield A(
+          1 + 2
+        )
+""", str(cm.exception))
 
     with self.assertRaises(_RuleVisitor.YieldVisitError) as cm:
       @rule(A, [])
       def g():
-        # This is a yield statement without an assignment.
+        # This is a yield statement without an assignment, and not at the end.
         yield Get(B, D, D())
         yield A()
     exc_msg = str(cm.exception)
```

Comment:
Let's maybe just `self.assertEquals` on the whole string? I suspect that would be a lot more clear, even if it's slightly more brittle...

Reply:
Remembering the number of times I rewrote this part and shuddering, yes, that sounds like a great idea.

Reply:
Done in 61fd025!